aten/src/ATen/native/ForeachOpsKernels.cpp

```diff
@@ -0,0 +1,23 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/DispatchStub.h>
+
+namespace at { namespace native {
+
+std::vector<Tensor> foreach_add_scalar_kernel_cpu(TensorList tensors, Scalar scalar) {
+  TORCH_CHECK(std::all_of(tensors.begin(), tensors.end(), [] (const Tensor& t) {
+    return t.layout() == at::kStrided;
+  }), "Only tensors with strided layouts are supported.");
+
+  TORCH_CHECK(std::all_of(tensors.begin(), tensors.end(), [] (const Tensor& t) {
+    return t.is_non_overlapping_and_dense();
+  }), "Only non overlapping and dense tensors are supported.");
```

Comment:
This sort of error checking will be common across all kernels.  Therefore, it belongs in the [multi_tensor_apply function itself](https://github.com/pytorch/pytorch/pull/41554/files#diff-ef767cf918d40173403532003ec78ac8R23), not in specific kernel functions.

The checks here are ok for scalar addition to a single tensor list, but not for multi tensor apply ops in which multiple tensor lists interact (e.g. adding a list of tensors to another list of tensors)

For ops where multiple tensor lists interact, what we want to check is:
- All tensors are kStrided.
- All tensors are non overlapping and dense.
- All TensorLists have the same length/number of tensors.
- For each tensor, its partner(s) in the other list(s) must have the same `size()` and `strides()`, ie
```C++
// list_of_tensor_lists[0] is the first list, list_of_tensor_lists[1] is the second list, etc.
// list_of_tensor_lists[0][0] is a Tensor that will interact with list_of_tensor_lists[1][0],
// list_of_tensor_lists[2][0], etc.
for (const TensorList& tl: list_of_tensor_lists {
  for (int i = 0; i < tl.size(); i++) {
    // checks every tensor against its partner in the first list
    tl[i].sizes() = list_of_tensor_lists[0][i].sizes();
    tl[i].strides() = list_of_tensor_lists[0][i].strides();
  }
}
```